eox_core/social_tpa_backends.py

```diff
@@ -61,3 +63,44 @@ def extra_data(self, user, uid, response, details=None, *args, **kwargs):  # pyl
         data["user_details"] = self.get_user_details(response)
 
         return data
+
+    def get_user_id(self, *args, **kwargs):
+        """
+        Insert a slug at the beginning of the user id. Retroactively add the
+        slug for social users that don't have one. The slug must be defined
+        in the 'OTHER_SETTINGS' field from the provider configuration.
+        In case a slug was not defined raise an exception due to misconfigured
+        backend.
+        This three behaviours can be controlled via the settings:
+            - 'SOCIAL_AUTH_NAMESPACED_UIDS': if false behave as originally
+            intended. If true, add a slug to the uid.
+            - 'SOCIAL_AUTH_ALLOW_SLUGLESS_UID': if false raise and exception when no slug
+            is defined. If true, return de original uid when no slug is defined.
+            - 'SOCIAL_AUTH_ALLOW_WRITE_SLUG_UID': If true update the uid of existing social
```

Comment:
I would suggest that this is always true and that we never want to "not update". As long as the "slug" field is created we want to update.

Reply:
The idea was that once all the old entries have been updated we can avoid a few queries to de DB since all the new entries will always have a slug if 'slug' is defined. But if you think it only adds complexity we can go with always true.